themes/finna2/templates/navibar.phtml

```diff
@@ -208,4 +208,12 @@
     </ul>
   <?php endif; ?>
 </div>
+<?php
+  $script = <<<JS
+    $(document).ready(function() {
+      finna.layout.initNavibar();
```

Comment:
Eikö tätä voisi kutsua suoraan finna-layout.js:n init-metodissa? inline-skriptiä tarvitaan yleensä vain sisällössä, jota ladataan dynaamisesti.

Reply:
Ja toisaalta, eikö navibar voisi itse hoitaa homman ilman js:ää?

Reply:
Tarkoitatko, että laittaisi jotain tällaista tms.?:
```php
<?php if ($url === $_SERVER['REQUEST_URI']): ?>
  <a aria-current="page" role="menuitem" <?= $target ? ' target="' . $this->escapeHtmlAttr($target) . '"' : ''?> href="<?=$this->escapeHtmlAttr($url)?>">
<?php else: ?>
  <a role="menuitem" <?= $target ? ' target="' . $this->escapeHtmlAttr($target) . '"' : ''?> href="<?=$this->escapeHtmlAttr($url)?>">
<?php endif; ?>
```
Näin ajattelin aluksi, mutta sitten pitää joka kohtaan laittaa ja meni mielestäni sotkuiseksi.

Reply:
Juu, paitsi ett tuon voisi laittaa inlinena:

`  <a<?=$url === $_SERVER['REQUEST_URI'] ? ' aria-current="page" : ''?> role="menuitem" <?= $target ? ' target="' . $this->escapeHtmlAttr($target) . '"' : ''?> href="<?=$this->escapeHtmlAttr($url)?>">
`

Mutta pitäisi varmaan siistä navibar.phml niin, että olisi vaikka yksi funktio, joka muodostaa kaikki linkit annettujen parametrien pohjalta, jotta ei tarvitsisi tehdä joka paikassa. Joka tapauksessa se olisi parempi ratkaisu kuin viilata jälikätetään js:llä.

Reply:
Okei jees kiitos, mä yritän siistiä tota navibar.phtml:ää siten, että saisin yhteen funktioon laitettua linkkien muodostamisen.

Reply:
Tutkailin tota navibar.phtml:ää ja se on aika haastava koostaa yhteen funktioon, koska siellä on niin paljon tarkistuksia yms. Mutta muutin nyt nuo aria current page attribuutit sinne ja poistin js:n.